jax/_src/lax/other.py

```diff
@@ -287,3 +287,35 @@ def _logaddexp_jvp(primals, tangents):
   tangent_out = lax.add(lax.mul(t1, lax.exp(lax.sub(_replace_inf(x1), _replace_inf(primal_out)))),
                         lax.mul(t2, lax.exp(lax.sub(_replace_inf(x2), _replace_inf(primal_out)))))
   return primal_out, tangent_out
+
+
+@custom_jvp
+def logaddexp2(x1: ArrayLike, x2: ArrayLike, /) -> Array:
+  """Compute log2(exp2(x1) + exp2(x2)) avoiding overflow."""
+  x1_arr = lax.asarray(x1)
+  x2_arr = lax.asarray(x2)
+  assert x1_arr.dtype == x2_arr.dtype
+
+  amax = lax.max(x1_arr, x2_arr)
+  invln2 = lax._const(amax, 1/np.log(2))
+  if dtypes.isdtype(x1_arr.dtype, "real floating"):
+    delta = lax.sub(x1_arr, x2_arr)
+    return lax.select(lax._isnan(delta),
+                      lax.add(x1_arr, x2_arr),  # NaNs or infinities of the same sign.
+                      lax.add(amax, lax.mul(invln2, lax.log1p(lax.exp2(lax.neg(lax.abs(delta)))))))
+  elif dtypes.isdtype(x1_arr.dtype, "complex floating"):
+    delta = lax.sub(lax.add(x1, x2), lax.mul(amax, lax._const(amax, 2)))
```

Comment:
Nit:
```suggestion
    delta = lax.sub(lax.add(x1_arr, x2_arr), lax.mul(amax, lax._const(amax, 2)))
```

Should there be sanity tests that would discover this?

Reply:
There are test failures that I suspect will be fixed by this change.

Reply:
@pearu I think I'll make the same correction in lax_other.logaddexp